caikit/config/config.py

```diff
@@ -89,7 +93,7 @@ def _update_global_config(cfg: aconfig.Config):
     # Update the config by merging the new updates over the existing config
     with _CONFIG_LOCK:
         # Locked just in case `configure()` is called concurrently for any reason
-        merge_configs(_CONFIG, cfg)
+        _CONFIG.update(cfg)
```

Comment:
> Is this function used anywhere else?

nope, but I did just simplify this to replace the ref instead of doing `update` and changed the comments / docstrings